### ISSUE TITLE
[wgsl-in] Missing built-in integer function implementations

### ISSUE THREAD

**Issue by hanawatson** [gfx-rs/naga#1824]
Hi - not sure if this is already known and on a to-do list somewhere, but while working on a WGSL fuzzing project I noticed naga has no implementation for a couple of built-in integer functions (https://gpuweb.github.io/gpuweb/wgsl/#integer-builtin-functions):

- countLeadingZeros
- countTrailingZeros

code example (analogous for countTrailingZeros):
```
@stage(compute) @workgroup_size(1)
fn compute_main() {
    var test_var = countLeadingZeros(53);
}
```

resultant error from attempting to compile with naga:
```
error: no definition in scope for identifier: 'countLeadingZeros'
  ┌─ test.wgsl:3:20
  │
3 │     var test_var = countLeadingZeros(53);
  │                    ^^^^^^^^^^^^^^^^^ unknown identifier

Could not parse WGSL
```

**Comment by teoxoy** [comment]
They are indeed missing, thanks for the report!

**Comment by fornwall** [comment]
@teoxoy This seems to have been fixed (#2226 and #2243).

**Comment by teoxoy** [comment]
Thanks for the reminder!
